dask_ml/model_selection/_search.py

```diff
@@ -1193,21 +1177,55 @@ def fit(self, X, y=None, groups=None, **fit_params):
         if scheduler is dask.threaded.get and n_jobs == 1:
             scheduler = dask.local.get_sync
 
-        out = scheduler(dsk, keys, num_workers=n_jobs)
+        if "Client" in type(getattr(scheduler, "__self__", None)).__name__:
```

Comment:
Will this approach be heavier on communication? Or is it the same amount, just spread out over time?

My real question is should we provide an option to disable this behavior?

Reply:
From what I can see I don't see why this would add additional overhead.  This "should" submit effectively the same task graph in the traditional dask scheduler as well as distributed.  The difference here is that the distributed scheduler allows you to submit the graph asynchronously and return back a bunch of futures with you can then yield the results as they return.  Given this I don't see why it would be preferable to disable this feature if a distributed client is present.